aws-api/src/main/java/com/amplifyframework/api/aws/auth/AuthRuleRequestDecorator.java

```diff
@@ -135,6 +139,17 @@ && userNotInReadRestrictingGroups(readAuthorizedGroupsMap, authType)) {
         return request;
     }
 
+    private boolean doesRuleAllowNonOwnerSubscribe(AuthRule authRule, AuthorizationType authMode) {
+        AuthorizationType typeForRule = AuthorizationType.from(authRule.getAuthProvider());
+        AuthStrategy strategy = authRule.getAuthStrategy();
+        List<ModelOperation> operations = authRule.getOperationsOrDefault();
+        return strategy != AuthStrategy.OWNER && strategy != AuthStrategy.GROUPS
+            && typeForRule != AuthorizationType.AMAZON_COGNITO_USER_POOLS
+            && typeForRule != AuthorizationType.OPENID_CONNECT
+            && typeForRule == authMode
+            && (operations.contains(ModelOperation.LISTEN) || operations.contains(ModelOperation.READ));
+    }
```

Comment:
Can this be simplified, or will equality checks make it easier to understand in the future?

```
strategy == AuthStrategy.PUBLIC
            && typeForRule == AuthorizationType.API_KEY
            && authMode == AuthorizationType.API_KEY
            && (operations.contains(ModelOperation.LISTEN) || operations.contains(ModelOperation.READ);
```

Reply:
🤔 this rewrite only covers a single case - public subscribe with an API key. Although that's the issue in the bug and likely the most common scenario, it's likely not the only one. I'll think about it a little more.

Reply:
I've updated to only address the case from the linked issue, when there is a public read rule and call is made using API_KEY. We can address other cases if other issues are opened in the future.